### ISSUE TITLE
Suppress mail notifications for "likes"

### ISSUE THREAD

**Issue by kr428** [friendica/friendica#12504]
At the moment, having Friendica configured to send out mail notifications for comments received on some post, this apparently also includes when a comment has been liked by some user.  A setting preventing e-mail notifications on "likes" would be nice. 🙂

**Comment by tobiasd** [comment]
On Wed, Dec 21, 2022 at 11:47:14PM -0800, kr428 wrote:
> At the moment, having Friendica configured to send out mail notifications for comments received on some post, this apparently also includes when a comment has been liked by some user.  A setting preventing e-mail notifications on "likes" would be nice. 🙂

Can you check your notification settings? You should be able to separately
activate 1st being notified about comments, and if you did so as 2nd stept if
you also want to be notified about likes (and again separately about re-shares).

**Comment by kr428** [comment]
Hmm, I'm unsure how it should be. At the moment it's set like this:

![notif](https://user-images.githubusercontent.com/685879/210199266-1b53d71d-dd97-48bf-a167-826745e73ba1.png)

The marked setting apparently(?) does change whether or not I get notifications for "likes" on posts in the web user interface. However, playing with these settings in various combinations, I never managed to change the behaviour regarding e-mail notifications for likes. Am I missing something here...?

**Comment by tobiasd** [comment]
Yes that's the setting. It *should* change the notification mail behavior as well, not only the notifications in the web interface.

**Comment by kr428** [comment]
Hmmm, in that case, it at least seems not to behave the way it should be.
(However, more than that, it would be helpful to have a different behaviour for mail and desktop notifications, at least for a personal use case: On the desktop it's interesting to see an ephemeral log of all interactions that happened. In e-mail, browsing through comments and having a quick pointer to access them is a neat way to not lose any "open ends"... but that's more a feature request I guess.)

**Comment by tobiasd** [comment]
There is an open ticket / plan for such a notification matrix in  #11337

**Comment by annando** [comment]
Can someone check if this is still an issue?

**Comment by kr428** [comment]
Still is, yes. (Still using procmail and friends to filter away "like" notifications.)

**Comment by annando** [comment]
@kr428 I've got two questions here:
1. Are you getting notifications by mail for events that you don't get notified at the desktop? Or do you receive this notification for likes also on the desktop?
2. Have you enabled the setting to get notified for likes?

**Comment by kr428** [comment]
![Screenshot_20231125-073837_Brave](https://github.com/friendica/friendica/assets/685879/5caf2407-4392-4f6c-a457-7bdc7ecbf703)
This is how it's configured here. I don't see "Like" notifications on the desktop, but I do receive e-mails for them. Same for reshared posts by the way.

**Comment by annando** [comment]
That's interesting. It should not happen when the option is not set. With PR https://github.com/friendica/friendica/pull/13671 I added some logging. After this is merged, please raise your log level to `debug` and wait for such mail to occur. Then have got a look at the log. You should have got an entry with `Send mail` in it.

**Comment by kr428** [comment]
@loma-one Could you by chance support here? Otherwise I'll need to spin an instance of my own...

**Comment by kr428** [comment]
This one still seems open. Still, I receive e-mail notifications for comments, likes and re-shares of my posts, regardless of what I configured in the Friendica web UI.

**Comment by kr428** [comment]
(Any input I can provide here to track this down...?)

**Comment by kr428** [comment]
Still broken in current Friendica version. Plus, I apparently can't really still filter "share" messages from replies. Likewise, a workaround - https://github.com/friendica/friendica/issues/13619 - also isn't fully possible; at the moment for e-mail notifications it generally seems to be all-or-nothing.

**Comment by foss-** [comment]
"Current" as in 2024.12 or 2025.07RC? Could you test RC (joinfriendica.de is an open signup instance running RC).

**Comment by kr428** [comment]
> "Current" as in 2024.12 or 2025.07RC? Could you test RC (joinfriendica.de is an open signup instance running RC).

Tested it on joinfriendica.de. It seems like this:

- By default, notifications for "likes" is disabled and these aren't sent out.
- As soon as I enable this, they are sent out via e-mail as expected.
- At this point, disabling this setting won't prevent these notifications from being sent out.

So it seems generally working but just toggling this setting doesn't?

**Comment by marcusxss** [comment]
It seems I can reproduce it on 2024.12 at least. The RC I can't really test currently.

> That's interesting. It should not happen when the option is not set. With PR [#13671](https://github.com/friendica/friendica/pull/13671) I added some logging. After this is merged, please raise your log level to `debug` and wait for such mail to occur. Then have got a look at the log. You should have got an entry with `Send mail` in it.

I enabled debug logs and got a relevant log line. Since it contains things like usernames and hostnames, I can send it in a PM on friendica or something, @annando ?

**Comment by kr428** [comment]
Seems fixed in current 2025.07RC. Thanks. 🙂
